### DISCUSSION TITLE
BREXX/370 V2R5M0

### DISCUSSION THREAD

**Discussion by mgrossmann** [mgrossmann/brexx370#97] [General]
See the [change log](https://github.com/mvslovers/brexx370/wiki/V2R5-Changelog) for a detailed list of improvements.

<hr /><em>This discussion was created from the release <a href='https://github.com/mvslovers/brexx370/releases/tag/V2R5M0'>BREXX/370 V2R5M0</a>.</em>

**Comment by dfgaeta** [comment]
Hi Mike,
I am Dan... We talked about BREXX some years ago, mainly we are trying to bring fullscreen features to in previous version. I am backing to play with BREXX and I am trying to build the current v2r5m0 but looks like there were some changes in package scripts. Is it correct? My intention is to update some points of interpreter to cover my personal purposes.
Thank you

**Comment by Peter-Jacob** [comment]
Hello Dan,
just out of curiosity, what exactly do you want to change? Maybe it's worth applying it to the general release. Concerning the package script, we have to wait for Mike`s response.

Peter

**Reply by mgrossmann** [reply]
Hi Dan,
Hi Peter,

I'm happy to have you here. :)

Just like Peter, I am of course interested in your changes. It would be super nice if you would send us a pull request with a short description of your changes and your reasons for them.

Regarding your question about the build process, where exactly is it stuck?

Best, Mike

**Comment by dfgaeta** [comment]
Hello Peter,

In your v2r3m0 I started writing an internal fullscreen panel management, something we have in ISPF host command processor. But the main point is the access to Sqlite, it was working fine, including store/fecth rexx variables based on sqlite column tables. It is a bunch of internal functions. I know to update this code is not a best practice, mainly if we are thinking to manage different versions.

I shared my code with Mr. Mike. There are lot of similar things in yours and my old package. I loved all things you guys have done. Awesome code.

Thank you!
Dan

**Reply by mgrossmann** [reply]
Hello Dan,

we had talked about SQLite back in the days, as well. 
I would still be interested in this topic. Do I remember correctly that you had ported SQLite using changes to Hercules?

We built our current implementation of FSS based on your code contribution back then. What we couldn't implement reliable and stable was reading the panel definitions from an external dataset. We later discarded that because we got a good alternative with Wally's ISPF. 

But in the end you have made a great contribution to BREXX/370. Thanks for that.

If you want, you are welcome to join our mainframe enthusiast chat, then we can exchange ideas in a short way.

https://discord.gg/gbn9pHrh

Mike

**Comment by dfgaeta** [comment]
Hey Mike,Thank you for your quick reply.Nice you remember our last emails and some Linkedin posts.. :)I am finishing some projects and now I have some tim eto back to playing with old IBM OSes. In past I worked in two different SQLite personal projects. The first one I added some code in general2.c hercules source. In summary I was intercepting a SVC routine to call SQLIte API. Basically I kept a connection to database per userid. The interesting point is related to performance.. It was fast and somehow, very stable... So.. there is a negative point, it is invasive... :(The similar SQLite project was based on a SQLite bash running in a TSO session. The engine was managing a DA dataset. But as you know, it is hard to work with this. I liked of performance and not so stable due to way to access data.The last and also interesting option is a DBMS managing VSAM clusters. Actually the current BREXX has all elements already available. VSAMIO, IKJ441CT (store/fetch rexx variables), Enq/Deq functions and TCPIP.. :)It is really nice to talk about this.. :)Thank you and keeping in touchDan GaetaOn Apr 26, 2022, at 8:04 AM, Mike Großmann ***@***.***> wrote:Hello Dan,we had talked about SQLite back in the days, as well. I would still be interested in this topic. Do I remember correctly that you had ported SQLite using changes to Hercules?We built our current implementation of FSS based on your code contribution back then. What we couldn't implement reliable and stable was reading the panel definitions from an external dataset. We later discarded that because we got a good alternative with Wally's ISPF.But in the end you have made a great contribution to BREXX/370. Thanks for that.If you want, you are welcome to join our mainframe enthusiast chat, then we can exchange ideas in a short way.https://discord.gg/gbn9pHrhMike—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
